scripts/firedrake-install

```diff
@@ -982,7 +982,10 @@ def clean(package):
     with directory(package):
         with environment(**blas):
             # loopy needs knowledge of the system BLAS just to run setup.py
-            check_call([python, "setup.py", "clean"])
+            if os.path.isfile("setup.py"):
+                check_call([python, "setup.py", "clean"])
+            else:  # PEP 518 build system
+                check_call(["rm", "-rf", "build"])
```

Comment:
I'm happy wit the first bit of logic here, if there's no `setup.py` we shouldn't try to call it. But what happens here if the build directory doesn't exist? (for instance if the installation of another package fails during firedrake-install, firedrake-update is used to resume the installation) There should probably be a try except block around the deletion of the build directory so the update doesn't fail here.

I think I'd also be a bit happier if we were using Python functionality to remove directories, it seems safer than subprocess calling `rm -rf` :grimacing: . I think `shutil.rmtree()` does what you want in this instance.

Reply:
Done, and force pushed.